src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncHelpers.CoreCLR.cs

```diff
@@ -49,81 +50,65 @@ public void Pop()
-        CORINFO_CONTINUATION_OSR_IL_OFFSET_IN_DATA = 4,
+        CORINFO_CONTINUATION_HAS_CONTINUATION_CONTEXT = 4,
+        CORINFO_CONTINUATION_HAS_RESULT = 8,
         // If this bit is set the continuation should continue on the thread
         // pool.
-        CORINFO_CONTINUATION_CONTINUE_ON_THREAD_POOL = 8,
+        CORINFO_CONTINUATION_CONTINUE_ON_THREAD_POOL = 16,
         // If this bit is set the continuation has a SynchronizationContext
         // that we should continue on.
-        CORINFO_CONTINUATION_CONTINUE_ON_CAPTURED_SYNCHRONIZATION_CONTEXT = 16,
+        CORINFO_CONTINUATION_CONTINUE_ON_CAPTURED_SYNCHRONIZATION_CONTEXT = 32,
         // If this bit is set the continuation has a TaskScheduler
         // that we should continue on.
-        CORINFO_CONTINUATION_CONTINUE_ON_CAPTURED_TASK_SCHEDULER = 32,
+        CORINFO_CONTINUATION_CONTINUE_ON_CAPTURED_TASK_SCHEDULER = 64,
     }
 
-    internal sealed unsafe class Continuation
+#pragma warning disable CA1852 // "Type can be sealed" -- no it cannot because the runtime constructs subtypes dynamically
+    internal unsafe class Continuation
     {
         public Continuation? Next;
-        public delegate*<Continuation, Continuation?> Resume;
-        public uint State;
+        public delegate*<Continuation, ref byte, Continuation?> Resume;
         public CorInfoContinuationFlags Flags;
+        public int State;
 
-        // Data and GCData contain the state of the continuation.
-        // Note: The JIT is ultimately responsible for laying out these arrays.
-        // However, other parts of the system depend on the layout to
-        // know where to locate or place various pieces of data:
-        //
-        // 1. Resumption stubs need to know where to place the return value
-        // inside the next continuation. If the return value has GC references
-        // then it is boxed and placed at GCData[0]; otherwise, it is placed
-        // inside Data at offset 0 if
-        // CORINFO_CONTINUATION_OSR_IL_OFFSET_IN_DATA is NOT set and otherwise
-        // at offset 4.
-        //
-        // 2. Likewise, Finalize[Value]TaskReturningThunk needs to know from
-        // where to extract the return value.
-        //
-        // 3. The dispatcher needs to know where to place the exception inside
-        // the next continuation with a handler. Continuations with handlers
-        // have CORINFO_CONTINUATION_NEEDS_EXCEPTION set. The exception is
-        // placed at GCData[0] if CORINFO_CONTINUATION_RESULT_IN_GCDATA is NOT
-        // set, and otherwise at GCData[1].
-        //
-        public byte[]? Data;
-        public object?[]? GCData;
-
-        public object GetContinuationContext()
+#if TARGET_64BIT
+        private const int POINTER_SIZE = 8;
+#else
+        private const int POINTER_SIZE = 4;
+#endif
+
+        private const int CONTINUATION_OFFSET_TO_DATA = POINTER_SIZE /* Next */ + POINTER_SIZE /* Resume */ + 8 /* Flags + State */;
```

Comment:
```suggestion
        private const int DataOffset = PointerSize /* Next */ + PointerSize /* Resume */ + 8 /* Flags + State */;
```
Follow C# naming convention?

Reply:
Fixed this